File: leetcode/algo2/day5/209_minimum_size_subarray_sum.cpp
#include <vector>
#include <iostream>
int minSubArrayLen(int target, std::vector<int> &nums)
{
    int left = 0;
    int res = INT_MAX;
    int sum = 0;
    for (int i = 0; i < nums.size(); i++)
    {
        sum += nums[i];
        while (sum >= target)
        {
            res = std::min(res, i - left + 1);
            sum -= nums[left++];
        }
    }
    if (res == INT_MAX)
        return 0;
    else
        return res;
}
int main()
{
    std::vector<int> nums = {2, 3, 1, 2, 4, 3};
    int target = 7;
    std::cout << minSubArrayLen(target, nums) << std::endl;
}

File: leetcode/algo2/day5/209_minimum_size_subarray_sum.js
var minSubArrayLen = function (target, nums) {
  let left = 0;
  let sum = 0;
  let res = Number.MAX_VALUE;

  for (let i = 0; i < nums.length; i++) {
    sum += nums[i];
    while (sum >= target) {
      res = Math.min(res, i - left + 1);
      sum -= nums[left++];
    }
  }
  return res === Number.MAX_VALUE ? 0 : res;
};
let nums = [2, 3, 1, 2, 4, 4];
target = 7;

console.log(minSubArrayLen(target, nums));
